relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -15,79 +15,63 @@ use crate::envelope::{Envelope, ItemType};
 /// Checks whether an event should be kept or removed by dynamic sampling.
 pub fn should_keep_event(
     sampling_context: Option<&DynamicSamplingContext>,
-    event: &Event,
+    event: Option<&Event>,
     ip_addr: Option<IpAddr>,
     project_state: &ProjectState,
+    trace_root_project_state: Option<&ProjectState>,
     processing_enabled: bool,
 ) -> SamplingResult {
-    let sampling_config = match &project_state.config.dynamic_sampling {
-        // without config there is not enough info to make up my mind
-        None => return SamplingResult::NoDecision,
-        Some(config) => config,
-    };
-
-    // when we have unsupported rules disable sampling for non processing relays
-    if !processing_enabled && sampling_config.has_unsupported_rules() {
-        return SamplingResult::Keep;
+    let mut matching_rule = None;
+
+    if let (Some(sampling_context), Some(trace_root_project_state)) =
```

Comment:
We should log the case where there's a sampling context with a project key but we couldn't resolve the project state. Since we no longer call this in the fast path, we would always expect a state. There are valid cases where this does not hold up of course, such as when the project is deleted or for some reason we're tracing across orgs. 

For now, a trace log should be fine.

Reply:
interesting case. even if the project is deleted i would expect a project state of some sorts.

>for some reason we're tracing across orgs.

even in this case processing relays have access to that other org, so that other org's sampling rules are getting applied